build_tools/update_environments_and_lock_files.py

```diff
@@ -437,13 +466,33 @@ def write_all_pip_lock_files(build_metadata_list):
         write_pip_lock_file(build_metadata)
 
 
-if __name__ == "__main__":
+@click.command()
+@click.option(
+    "--select-build",
```

Comment:
I found this quite convenient when only generating a subset of all the possible lock files, regenerating all the lock files can take a while (~4 minutes on my machine)

Reply:
This does add `click` as direct dependency. `click` is already installed because is a dependency for `pip-tools`. Let's add it into the list of dependencies?

https://github.com/scikit-learn/scikit-learn/blob/ef2ef3176652aed48312998c9cce668bbd5d3e8d/build_tools/update_environments_and_lock_files.py#L25-L29

Reply:
`click` is a indirect dependency right? `conda-lock` (and `pip-tools` actually as you said) depends on `click`, so I thought it was not necessary to mention it.

Reply:
`jinja2` was included in the list as a indirect dependency from `conda-lock`. (I'm trying to be consistent with what goes in the list)

Reply:
I removed `jinja2` from the list since it is a dependency of `conda-lock`